APMrover2/GCS_Mavlink.cpp

```diff
@@ -529,7 +529,13 @@ MAV_RESULT GCS_MAVLINK_Rover::_handle_command_preflight_calibration(const mavlin
             return MAV_RESULT_FAILED;
         }
     } else if (is_equal(packet.param6, 1.0f)) {
-        if (rover.g2.windvane.start_calibration()) {
+        if (rover.g2.windvane.start_direction_calibration()) {
+            return MAV_RESULT_ACCEPTED;
+        } else {
+            return MAV_RESULT_FAILED;
+        }
+    } else if (is_equal(packet.param7, 1.0f)) {
```

Comment:
I'm not sure if this is a packet.param7 is a valid/sensible thing to use here?

Reply:
alternatively we could use param6 = "1" for direction calibration, "2" for wind vane speed calibration..

Reply:
that seems more sensible, matches the pram then, i have changes